homeassistant/components/insteon/config_flow.py

```diff
@@ -0,0 +1,353 @@
+    return new_options, housecode, unitcode
+
+
+class InsteonFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Insteon config flow handler."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Define the config flow to handle options."""
+        return InsteonOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """For backward compatibility."""
+        return await self.async_step_init(user_input=user_input)
+
+    async def async_step_init(self, user_input=None):
+        """Init the config flow."""
+        errors = {}
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        if user_input is not None:
+            plm = 1 if user_input.get(STEP_PLM) else 0
+            hubv2 = 1 if user_input.get(STEP_HUB_V2) else 0
+            hubv1 = 1 if user_input.get(STEP_HUB_V1) else 0
+            if _only_one_selected(plm, hubv1, hubv2):
+                if plm:
+                    return await self.async_step_plm()
+                if hubv1:
+                    return await self.async_step_hubv1()
+                if hubv2:
+                    return await self.async_step_hubv2()
+            errors["base"] = "select_single"
+        data_schema = vol.Schema(
+            {
+                vol.Optional(STEP_PLM): bool,
+                vol.Optional(STEP_HUB_V2): bool,
+                vol.Optional(STEP_HUB_V1): bool,
+            }
+        )
+        return self.async_show_form(
+            step_id="init", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_plm(self, user_input=None):
+        """Set up the PLM modem type."""
+        errors = {}
+        if user_input is not None:
+            if await _async_connect(**user_input):
+                return self.async_create_entry(title="", data=user_input)
+            errors["base"] = "cannot_connect"
+        schema_defaults = user_input if user_input is not None else {}
+        data_schema = build_plm_schema(**schema_defaults)
+        return self.async_show_form(
+            step_id=STEP_PLM, data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_hubv1(self, user_input=None):
+        """Set up the Hub v1 modem type."""
+        errors = {}
+        if user_input is not None:
+            user_input[CONF_HUB_VERSION] = 1
+            if await _async_connect(**user_input):
+                return self.async_create_entry(title="", data=user_input)
+            user_input.pop(CONF_HUB_VERSION)
+            errors["base"] = "cannot_connect"
+        schema_defaults = user_input if user_input is not None else {}
+        data_schema = build_hub_v1_schema(**schema_defaults)
+        return self.async_show_form(
+            step_id=STEP_HUB_V1, data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_hubv2(self, user_input=None):
```

Comment:
This code is substantial similar to `async_step_hubv1`

Can you make this more DRY?